src/border_agent/border_agent.cpp

```diff
@@ -388,6 +390,16 @@ bool BorderAgent::IsThreadStarted(void) const
     return role == OT_DEVICE_ROLE_CHILD || role == OT_DEVICE_ROLE_ROUTER || role == OT_DEVICE_ROLE_LEADER;
 }
 
+std::string BorderAgent::BaseServiceInstanceName() const
+{
+    const otExtAddress *extAddress = otLinkGetExtendedAddress(mNcp.GetInstance());
+    std::stringstream   ss;
+
+    ss << kBorderAgentServiceInstanceName << " " << std::uppercase << std::hex << std::setfill('0') << std::setw(2)
```

Comment:
Is it better to add a separator like '#' between the name and the discriminator? Say `OpenThread #ABCD`.